src/index.ts

```diff
@@ -107,19 +108,36 @@ export const DEFAULT_OPTIONS: Options = {
   unknownAny: true,
 }
 
+function isYml(filename: string) {
+  return filename.split('.').pop() === 'yml'
+}
+
 export function compileFromFile(filename: string, options: Partial<Options> = DEFAULT_OPTIONS): Promise<string> {
   const contents = Try(
     () => readFileSync(filename),
     () => {
       throw new ReferenceError(`Unable to read file "${filename}"`)
     },
   )
-  const schema = Try<JSONSchema4>(
-    () => JSON.parse(contents.toString()),
-    () => {
-      throw new TypeError(`Error parsing JSON in file "${filename}"`)
-    },
-  )
+
+  let schema: JSONSchema4
+
+  if (isYml(filename)) {
+    schema = Try<JSONSchema4>(
+      () => JSON.parse(JSON.stringify(yaml.load(contents.toString()))),
```

Comment:
Can you remove the `JSON.parse(JSON.stringify())`?

Reply:
Updated as suggested